src/pkg/packager/remove.go

```diff
@@ -114,3 +89,44 @@ func (p *Packager) updatePackageSecret(deployedPackage types.DeployedPackage, se
 		message.Warnf("Unable to update the %s package secret: %#v", secretName, err)
 	}
 }
+
+func (p *Packager) removeComponent(components []types.ZarfComponent, deployedComponent types.DeployedComponent, spinner *message.Spinner) error {
+	c := utils.Find(components, func(t types.ZarfComponent) bool {
```

Comment:
Because `OnRemove` is new functionality I would rather add `Actions.OnRemove` to `types.DeployedComponent`.  A missing key should initialize to the zero value (empty array) which would be fine for any existing deployed components without `OnRemove` actions.

Reply:
For my 2-cents;

I kind of like how the `types.Deployed{Package}/{Component}/{Chart}` all have the minimal amount of information necessary to know what is actually deployed for that type. At the moment it's purely state based information.

Adding a OnRemove lifecycle hook descriptor to components seems to be a bit overloady.

Reply:
I can see that.  I'll mark as resolved even though the loop still feels weird.